File: lib/models/user_notification.dart
/// One row in Sarah's bell inbox.
///
/// Type values seen today:
///   * `post_liked`        — actor liked one of my posts
///   * `post_commented`    — actor left a comment on one of my posts
class UserNotification {
  final int id;
  final int userId;
  final int? actorId;
  final String? actorName;
  final String type;
  final int? postId;
  final int? commentId;
  final String? bodySnippet;
  final DateTime? readAt;
  final DateTime createdAt;

  const UserNotification({
    required this.id,
    required this.userId,
    required this.type,
    required this.createdAt,
    this.actorId,
    this.actorName,
    this.postId,
    this.commentId,
    this.bodySnippet,
    this.readAt,
  });

  bool get unread => readAt == null;

  factory UserNotification.fromJson(Map<String, dynamic> json) =>
      UserNotification(
        id: (json['id'] as num).toInt(),
        userId: (json['userId'] as num).toInt(),
        actorId: (json['actorId'] as num?)?.toInt(),
        actorName: json['actorName'] as String?,
        type: json['type'] as String? ?? '',
        postId: (json['postId'] as num?)?.toInt(),
        commentId: (json['commentId'] as num?)?.toInt(),
        bodySnippet: json['bodySnippet'] as String?,
        readAt: DateTime.tryParse(json['readAt'] as String? ?? ''),
        createdAt: DateTime.tryParse(json['createdAt'] as String? ?? '') ??
            DateTime.now(),
      );
}

File: backend/internal/models/user_notification.go
package models

import "time"

// UserNotification — one row in user_notifications. Drives Sarah's bell
// inbox. Type values: "post_liked", "post_commented", "subscription_active",
// "subscription_pending", etc. The Flutter screen renders type-specific
// copy from these primitives.
type UserNotification struct {
	ID           int64      `json:"id"`
	UserID       int64      `json:"userId"`
	ActorID      *int64     `json:"actorId,omitempty"`
	ActorName    *string    `json:"actorName,omitempty"`
	Type         string     `json:"type"`
	PostID       *int64     `json:"postId,omitempty"`
	CommentID    *int64     `json:"commentId,omitempty"`
	BodySnippet  *string    `json:"bodySnippet,omitempty"`
	ReadAt       *time.Time `json:"readAt,omitempty"`
	CreatedAt    time.Time  `json:"createdAt"`
}
